packages/peregrine/lib/state/Checkout/state.js

```diff
@@ -0,0 +1,229 @@
+import { useCallback, useMemo, useReducer } from 'react';
```

Comment:
You'll note in here that I didn't copy over _everything_. For example I left out much of the reducer functions that were "enclosing" actions like `begin` and `reject`. If we want to store "error" or "active" state such as `editing` or `step` or `submitting` we can, but IMO those belong in the UI.